source/dofs/dof_handler_policy.cc

```diff
@@ -2268,23 +2272,31 @@ namespace internal
         const unsigned int
         n_cpus = Utilities::MPI::n_mpi_processes (tr->get_communicator());
 
-        unsigned int n_levels = Utilities::MPI::max(dof_handler.get_triangulation().n_levels(), tr->get_communicator());
-
+        // loop over all levels that exist globally (across all
+        // processors), even if the current processor does not in fact
+        // have any cells on that level or if the local part of the
+        // Triangulation has fewer levels. we need to do this because
+        // we need to communicate across all processors on all levels
+        const unsigned int n_levels = Utilities::MPI::max(dof_handler.get_triangulation().n_levels(),
+                                                          tr->get_communicator());
```

Comment:
Isn't there `Triangulation::n_global_levels()` that avoids collecting that information with an `Allreduce`?